hypothesis-python/src/hypothesis/_settings.py

```diff
@@ -457,7 +457,17 @@ def __repr__(self):
 
     @classmethod
     def all(cls) -> List["HealthCheck"]:
-        return list(HealthCheck)
+        attributes = list(HealthCheck)
+        strict_errors = [HealthCheck.return_value, HealthCheck.not_a_test_method]
+
+        for strict_error in strict_errors: 
+            attributes.remove(strict_error)
+
+        return attributes
+    
+    # @classmethod
+    # def __iter__(cls) -> Iterator["HealthCheck"]:
+    #     return iter(cls.all())
```

Comment:
Let's keep the existing implementation of `.all()`, and define `__iter__` using eg

```python
def __iter__(self):
    return iter(x for x in super().__iter__() if x.name not in self.__deprecated_names)
```

Then we can create the `__deprecated_names` attribute as a tuple of strings, and add:

```python
def __getattr__(self, name):
    if name in self.__deprecated_names:
        note_deprecation(...)
    return super().__getattr__(name)
```

Haven't checked any of this, but I think it'll work, and should fix most of the test failures.